apps/prairielearn/src/lib/db-types.ts

```diff
@@ -7,7 +15,25 @@ import { z } from 'zod';
 // string is actually a number. If it's not, we want to fail quickly.
 export const IdSchema = z
   .string({ coerce: true })
-  .refine((val) => /^\d+$/.test(val), { message: 'ID is not an integer' });
+  .refine((val) => /^\d+$/.test(val), { message: 'ID is not a non-negative integer' });
+
+export const IntervalSchema = z.string().transform((val) => {
+  const interval = parsePostgresInterval(val);
+
+  // This calculation matches Postgres's behavior when computing the number of
+  // milliseconds in an interval with `EXTRACT(epoch from '...'::interval) * 1000`.
+  // The noteworthy parts of this conversion are that 1 year = 365.25 days and
+  // 1 month = 30 days.
```

Comment:
I wasn't actually able to find this documented explicitly in the Postgres docs, but I found references to this fact [on StackOverflow](https://stackoverflow.com/questions/952493/how-do-i-convert-an-interval-into-a-number-of-hours-with-postgres) and confirmed this experimentally.